CMakeLists.txt

```
@@ -64,59 +65,42 @@ endif(DOXYGEN_FOUND)
 
 ################################
 # Create unit test excecutable #
-find_package(GTest QUIET)
-if(${GTest_FOUND})
-	# Gtest on System Installed
-	message("GTest: Installed using GTest")
-	include(GoogleTest)
-else()
+option(LIBJAPI_ENABLE_TESTING "Enable testing with googletest" ON)
+if(LIBJAPI_ENABLE_TESTING)
+  enable_testing()
+
   # Include GoogleTest in the project
   include(FetchContent)
   FetchContent_Declare(
     googletest
     GIT_REPOSITORY https://github.com/google/googletest.git
-    GIT_TAG release-1.8.1
-  )
+    GIT_TAG release-1.8.0)
   FetchContent_MakeAvailable(googletest)
   include(GoogleTest)
-endif()
-    
-# Define test executable
-add_executable(testsuite
-  EXCLUDE_FROM_ALL
-    test/japi_test.cc
-)
 
-target_compile_options(testsuite PUBLIC "-pthread")
+  # Define test executable
+  add_executable(testsuite test/japi_test.cc)
 
-target_link_libraries(testsuite
-  gtest_main
-  japi
-)
+  target_compile_options(testsuite PUBLIC "-pthread")
 
-# Include directories
-target_include_directories(testsuite
-  PUBLIC
-    include/
-    src/
-    ${JSONC_INCLUDE_DIRS}
-)
+  target_link_libraries(testsuite gtest_main japi)
 
-add_custom_target(run_test COMMAND testsuite DEPENDS testsuite)
+  # Include directories
+  target_include_directories(testsuite PUBLIC include/ src/
+                                              ${JSONC_INCLUDE_DIRS})
 
+  gtest_discover_tests(testsuite)
 ################################
 # Test code coverage #
 
-if(CMAKE_BUILD_TYPE MATCHES "Debug")
   list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake")
   include(${CMAKE_SOURCE_DIR}/cmake/CodeCoverage.cmake)
   set(COVERAGE_EXCLUDES
       "doxydir"
       "/usr/include/*"
-      "googletest/*"
       "test/*"
   )
   append_coverage_compiler_flags()
 
   setup_target_for_coverage_lcov(NAME coverage EXECUTABLE testsuite)
```

```suggestion

  if(CMAKE_BUILD_TYPE MATCHES "Debug")
    list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake")
    include(${CMAKE_SOURCE_DIR}/cmake/CodeCoverage.cmake)
    set(COVERAGE_EXCLUDES
        "doxydir"
        "/usr/include"
        "_deps/googletest"
        "test/*"
    )
    append_coverage_compiler_flags()

    setup_target_for_coverage_lcov(NAME coverage EXECUTABLE testsuite)
  endif()
```

googletest still needs to be excluded from coverage.
Fully testing on Mac was not possible. @ndml28 can you quicly test on Debian before merging?

---

Okay done! should be fine now.